.github/workflows/ci.yml

```
@@ -43,6 +43,43 @@ jobs:
       - uses: DeterminateSystems/magic-nix-cache-action@v4
       - run: nix ${{ matrix.command }}
 
+  cargo:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: "write" # added for magic-nix-cache-action
+      contents: "read"  # added for magic-nix-cache-action
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - command: check --locked --all
+          - command: clippy --locked --all -- -D warnings
+          - command: fmt --all -- --check
+          - command: test --locked --all
+          - command: test --no-default-features --locked --all
+          - command: test --all-features --locked --all -- --nocapture
+    steps:
+    - uses: actions/checkout@v3
+    - uses: DeterminateSystems/nix-installer-action@v10
+    - uses: DeterminateSystems/magic-nix-cache-action@v4
+    - uses: Swatinem/rust-cache@v2
+    - env:
+        TMPDIR: ${{ runner.temp }}
+      run: nix develop --command cargo ${{ matrix.command }}
```

```suggestion
      run: nix develop .#essential-rust-app-dev --command cargo ${{ matrix.command }}
```
Will run a lot faster and save a bunch of CI time